.buildkite/publish/dra/init_dra_publishing.sh

```diff
@@ -0,0 +1,153 @@
+    exit 2
+  fi
+
+  export VAULT_ROLE_ID="$(echo $DRA_CREDS | jq -r '.role_id')"
+  if [[ "${VAULT_ROLE_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  export VAULT_SECRET_ID="$(echo $DRA_CREDS | jq -r '.secret_id')"
+  if [[ "${VAULT_SECRET_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  unset DRA_CREDS
+  set -x
+}
+
+# function to unset the vault credentials.
+function unsetDraVaultCredentials() {
+  set +x
+  unset VAULT_ROLE_ID
+  unset VAULT_SECRET_ID
+
+  # and reset our original vault address
+  export VAULT_ADDR="${ORIGINAL_VAULT_ADDR:-}"
+
+  set -x
+}
+
+# function to generate dependency report.
+function generateDependencyReport() {
+  echo "name,version,url,license,sourceURL" > $1 # TODO, do this right
+}
+
+# generate the dependency report and publish SNAPSHOT artifacts
+if [[ "${PUBLISH_SNAPSHOT:-}" == "true" ]]; then
+  # note that this name is set by infra and must be in the format specified
+  dependencyReportName="dependencies-${VERSION}-SNAPSHOT.csv";
+  echo "-------- Generating SNAPSHOT dependency report: ${dependencyReportName}"
+  generateDependencyReport $DEPENDENCIES_REPORTS_DIR/$dependencyReportName
+
+  echo "-------- Publishing SNAPSHOT DRA Artifacts"
+  cp $RELEASE_DIR/dist/elasticsearch_connectors-${VERSION}.zip $RELEASE_DIR/dist/dra-artifacts/connectors-${VERSION}-SNAPSHOT.zip
+  setDraVaultCredentials
+  export WORKFLOW="snapshot"
+
+  # TODO - when ready, remove the gate and ungate test vars
+  if [[ "${DRA_PUBLISH_GATE:-}" == "" || "${DRA_UNGATE_TEST:-}" == "true" ]]; then
+    source "${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh"
+  else
+    echo "DRA_PUBLISH_GATE is locked. This is where we would call 'source ${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh'"
+  fi
+  unsetDraVaultCredentials
+  rm -rf "${DEPENDENCIES_REPORTS_DIR}/*"
+fi
+
+# generate the dependency report and publish STAGING artifacts
+if [[ "${PUBLISH_STAGING:-}" == "true" ]]; then
+  dependencyReportName="dependencies-${VERSION}.csv";
+  echo "-------- Generating STAGING dependency report: ${dependencyReportName}"
+  generateDependencyReport $DEPENDENCIES_REPORTS_DIR/$dependencyReportName
+
+  echo "-------- Publishing STAGING DRA Artifacts"
+  cp $RELEASE_DIR/dist/elasticsearch_connectors-${VERSION}.zip $RELEASE_DIR/dist/dra-artifacts/connectors-${VERSION}.zip
+  setDraVaultCredentials
+  export WORKFLOW="staging"
+
+  # TODO - when ready, remove the gate var
+  if [[ "${DRA_PUBLISH_GATE:-}" == "" ]]; then
+    source "${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh"
+  else
+    echo "DRA_PUBLISH_GATE is locked. This is where we would call 'source ${PROJECT_ROOT}/.buildkite/publish/dra/publish-daily-release-artifact.sh'"
+  fi
+  unsetDraVaultCredentials
+  rm -rf "${DEPENDENCIES_REPORTS_DIR}/*"
```

Comment:
You can probably always run it after script exits like [here](https://stackoverflow.com/questions/15656492/writing-try-catch-finally-in-shell)?

Reply:
TIL about `trap`. However, reading about it, I don't think this is what we want. We need to clean up the SNAPSHOT dependency report file before we move on to the staging dependency reporting. We can't wait until exit.

And since we have `set -e`, there's no risk of something going wrong in producing the SNAPSHOT, leaving that dependency CSV, but progressing to the staging logic.

Reply:
If you want to be extra safe, adding this (and the `unset`) into a `trap` command on exit is best as discussed, but since the Buildkite container will be removed after completion, I think it's fine and not a huge concern...

Reply:
> We need to clean up the SNAPSHOT dependency report file before we move on to the staging dependency reporting. We can't wait until exit.

Ah - my bad - I didn't notice that... carry on :)